packages/nx/src/utils/find-matching-projects.ts

```diff
@@ -167,6 +167,17 @@ function addMatchingProjectsByName(
   }
 
   if (!isGlobPattern(pattern.value)) {
+    const regex = new RegExp(`\\b${pattern.value}\\b`);
```

Comment:
Apparently `\b` does not include underscores... should we make it include `_` as a boundary?

Rust projects in particular have a lot of `_`

Reply:
Yes, we should keep `_`. I'll udpate it.